lib/optimizely/odp/odp_segment_manager.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2022, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'optimizely/logger'
+require_relative 'zaius_graphql_api_manager'
+
+module Optimizely
+  class OdpSegmentManager
+    # Schedules connections to ODP for audience segmentation and caches the results
+    attr_reader :odp_config, :segments_cache, :zaius_manager, :logger
+
+    def initialize(cache_size, cache_timeout_in_secs, odp_config, api_manager = nil, logger = nil, proxy_config = nil)
+      @odp_config = odp_config
+      @logger = logger || NoOpLogger.new
+      @zaius_manager = api_manager || ZaiusGraphQLApiManager.new(logger: @logger, proxy_config: proxy_config)
+      @segments_cache = Optimizely::LRUCache.new(cache_size, cache_timeout_in_secs)
```

Comment:
do we want an ability to pass an implementation from outside?

Reply:
yea good catch